engine/api/ascode/sync/sync.go

```diff
@@ -38,61 +39,58 @@ func SyncAsCodeEvent(ctx context.Context, db *gorp.DbMap, store cache.Store, pro
 		}
 	}
 
-	asCodeEvents, err := ascode.LoadAsCodeEventByRepo(ctx, db, fromRepo)
+	tx, err := db.Begin()
 	if err != nil {
-		return nil, fromRepo, err
+		return nil, fromRepo, sdk.WithStack(err)
 	}
+	defer tx.Rollback() //nolint
 
-	tx, err := db.Begin()
+	asCodeEvents, err := ascode.LoadAsCodeEventByRepo(ctx, tx, fromRepo)
 	if err != nil {
-		return nil, fromRepo, sdk.WrapError(err, "unable to start transaction")
+		return nil, fromRepo, err
 	}
-	defer tx.Rollback() //nolint
 
 	eventLeft := make([]sdk.AsCodeEvent, 0)
-	eventDeleted := make([]sdk.AsCodeEvent, 0)
+	eventToDelete := make([]sdk.AsCodeEvent, 0)
 	for _, ascodeEvt := range asCodeEvents {
+		pr, err := client.PullRequest(ctx, app.RepositoryFullname, int(ascodeEvt.PullRequestID))
+		if err != nil && !sdk.ErrorIs(err, sdk.ErrNotFound) {
+			return nil, fromRepo, sdk.WrapError(err, "unable to check pull request status")
+		}
+		prNotFound := sdk.ErrorIs(err, sdk.ErrNotFound)
 
-		merged, closed, err := CheckPullRequestStatus(ctx, client, app.RepositoryFullname, ascodeEvt.PullRequestID)
-		if err != nil {
-			return nil, fromRepo, err
+		if prNotFound {
+			log.Debug("Pull request %s #%d not found", app.RepositoryFullname, int(ascodeEvt.PullRequestID))
 		}
+
```

Comment:
How do you manage now, workflow that need an update of from_repository ( PR created for migrate a workflow to as code workflow ) ?

Reply:
Move lower in the code